scripts/prepare_bindings.py

```diff
@@ -277,11 +277,11 @@ def main(args):
         if options.find_swig:
```

Comment:
Why not just ignore the swig handling if static bindings are in use?  That is `--find-swig` is meaningless.  Simply hoist the check above the `if options.find_swig`.

Reply:
That would break the scenario where you are okay with using the static bindings if swig doesn't exist, but you still want to try and find it. I'm honestly not sure if that's a scenario that we need to support, but it's what the script does currently.

Reply:
Hmm, I see.  Personally, I think that is something we should explicitly prohibit.  You select between the static or dynamic behaviour.  Using swig if available makes it unpredictable.  What happens if I uninstalled swig and forgot about that?  Suddenly the build switches behaviour.  But, if there is a strong reason to support that, then, sure, the change LGTM.